src/main/java/com/redhat/devtools/lsp4ij/features/semanticTokens/viewProvider/LSPSemanticToken.java

```diff
@@ -134,6 +139,7 @@ class LSPSemanticToken {
     private final boolean isFileLevel;
     private final LSPSemanticTokenElementType elementType;
     private volatile LSPSemanticTokenPsiElement element = null;
+    private final ThreadLocal<Integer> lastRequestedOffsetPtr = new InheritableThreadLocal<>();
```

Comment:
This is one of the two inheritable thread locals that could contain the last requested offset into the file that didn't correspond to a semantic token element. It's automatically cleared when a semantic token element is requested so that it stays "fresh". This effectively allows us to simulate having a fully-populated PSI element structure for the file while only having PSI elements for semantic tokens.